File: algo-method.com/tasks/1069/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];
const print = (...messages: any) => console.log(...messages);

const int2bin = (num: number, width: number): string =>
  num.toString(2).padStart(width, "0");

const X = input().split(" ").map(Number);

const getSquare = (binX: string, j: number): string => {
  if (binX[2 * j] === "0" && binX[2 * j + 1] === "0") return ".";
  if (binX[2 * j] === "0" && binX[2 * j + 1] === "1") return "o";
  return "x";
};

const getLine = (x: number): string => {
  const binX = int2bin(x, 16);
  return [...Array(8)].map((_, j) => getSquare(binX, j)).join("");
};

const ans = X.map(getLine);
print(ans.join("\n"));

File: algo-method.com/tasks/1069/main.py
def int2bin(number: int, width: int) -> str:
    return bin(number)[2:].zfill(width)


X = list(map(int, input().split()))


def get_square(bin_x: str, j: int) -> str:
    if bin_x[2 * j] == "0" and bin_x[2 * j + 1] == "0":
        return "."
    if bin_x[2 * j] == "0" and bin_x[2 * j + 1] == "1":
        return "o"
    return "x"


def get_line(x: int) -> str:
    bin_x = int2bin(x, 16)
    return "".join(get_square(bin_x, j) for j in range(8))


ans = map(get_line, X)
print(*ans, sep="\n")
